tls/s2n_ktls_io.c

```diff
@@ -435,6 +481,19 @@ int s2n_sendfile(struct s2n_connection *conn, int in_fd, off_t offset, size_t co
 #endif
 
     *blocked = S2N_NOT_BLOCKED;
+
+    /* Because we pass the input file descriptor to the kernel without examining
+     * it, we don't know how many bytes actually need to be sent. We therefore
+     * can't verify that the send is safe with respect to the encryption limit
+     * before sending the records. Instead, we raise a fatal error afterwards if
+     * the send violated the encryption limit.
+     *
+     * An application should treat S2N_ERR_KTLS_KEY_LIMIT as a very high severity
```

Comment:
I feel like this paragraph belongs in a doc that the user is actually going to read.